src/lib.rs

```diff
@@ -144,12 +144,13 @@ pub fn desugar_book(
   if !ctx.info.has_errors() { Ok(ctx.info) } else { Err(ctx.info) }
 }
 
-pub fn run_book(
+pub fn run_book_with_fn(
   mut book: Book,
   run_opts: RunOpts,
   compile_opts: CompileOpts,
   diagnostics_cfg: DiagnosticsConfig,
   args: Option<Vec<Term>>,
+  run_fn: impl Fn(&str) -> Result<Output, String>,
```

Comment:
Instead of passing a function, isn't passing the name of the hvm command ("run", "run-c", etc) enough?